src/Auth0Client.ts

```diff
@@ -25,24 +25,39 @@ import version from './version';
 const lock = new Lock();
 const GET_TOKEN_SILENTLY_LOCK_KEY = 'auth0.lock.getTokenSilently';
 
+const cacheStrategies = {
+  memory: () => new InMemoryCache(),
+  localstorage: () => new LocalStorageCache()
+};
+
 /**
  * Auth0 SDK for Single Page Applications using [Authorization Code Grant Flow with PKCE](https://auth0.com/docs/api-auth/tutorials/authorization-code-grant-pkce).
  */
 export default class Auth0Client {
-  private cache: Cache;
+  private cache: ICache;
   private transactionManager: TransactionManager;
   private domainUrl: string;
   private tokenIssuer: string;
   private readonly DEFAULT_SCOPE = 'openid profile email';
 
+  cacheStrategy: string;
```

Comment:
The TypeScript way here would be to define an enum for the cacheStrategy with the available options.  e.g.

```typescript
enum CacheStrategy { Memory, LocalStorage };
```

Additionally I wonder if instead of strategy we should call this cache persistence?  Strategy is quite vague and could encompass things like how long it is cached for.

Reply:
I'm not married to the name. How about `cacheMode`?

Regarding the enum, that would be good and it looks like I can still specify string enums too:

```js
enum CacheMode {
  Memory = 'memory',
  LocalStorage = 'localstorage'
}
```

Just thinking about those users who are not using our library with TypeScript but just normal JavaScript. Would they still be able to reference the option in a non-typed form? It might be important seeing is we don't currently export the TypeScript types (but we should do).

Reply:
I went with 'cacheLocation` for this.

Also, I will implement the enum in a separate PR when #310 comes in. Right now if I try to export a type from the place I want I get a load of errors, but this should be solved by that PR.